ospfd/ospf_interface.c

```diff
@@ -804,6 +805,10 @@ int ospf_if_up(struct ospf_interface *oi)
 int ospf_if_down(struct ospf_interface *oi)
 {
 	struct ospf *ospf;
+	struct route_node *rn;
```

Comment:
so ... can you say more about what goes wrong - does zebra unilaterally remove/discard some of ospf's existing routes? why is it bad not to react to a series of events like this?

Reply:
zebra is not unilateraly removing anything.

order of events that I am seeing in the log:

interface up event 
ospf runs spf sends route install to zebra
interface down event *Before* the route install is processed
route install is processed can't install it because the nexthop is no good ( interface down )
interface up event 
ospf finally gets around to running spf.  Nothing has changed from it's perspective and it doesn't reinstall the route

What I do is that if we receive a interface down event I look through all routes that we have and if any of them are using that ifindex that is going down, we mark the route as changed

Then after spf is run and we are checking to see if we should install the route, ospf doesn't see any changes so it doesn't send anything to zebra.

zebra now has a route with an inactive nexthop.

Reply:
ah, ok, got it - I was imagining a slightly different sequence, thanks.